apis/falcon/v1alpha1/falconcontainer_types.go

```diff
@@ -1,53 +1,123 @@
 	FalconAPI FalconAPI `json:"falcon_api"`
+
 	// Registry configures container image registry to which the Falcon Container image will be pushed
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Falcon Container Image Registry Configuration",order=2
 	Registry RegistrySpec `json:"registry,omitempty"`
 
-	// InstallerArgs are passed directly down to the Falcon Container Installer. Users are advised to consult Falcon Container documentation to learn about available command line arguments at https://falcon.crowdstrike.com/documentation/146/falcon-container-sensor-for-linux
-	InstallerArgs []string `json:"installer_args,omitempty"`
 	// Injector represents additional configuration for Falcon Container Injector
-	Injector *FalconContainerInjectorSpec `json:"injector,omitempty"`
+	// +kubebuilder:default:={imagePullPolicy:Always}
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Falcon Container Injector Configuration",order=3
+	Injector FalconContainerInjectorSpec `json:"injector,omitempty"`
 
-	// Falcon Container Version. The latest version will be selected when version specifier is missing.
+	// Falcon Container Version Pinning. If not set to false, once a sensor version is set, it is used until manually adjusted.
+	// +kubebuilder:default:=true
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Falcon Container Image Version Pinning",order=4
+	VersionPinning bool `json:"versionPinning,omitempty"`
+
+	// +kubebuilder:validation:Pattern="^.*:.*$"
+	// +operator-sdk:cv:customresourcedefinitions:type=spec,displayName="Falcon Container Image URI",order=5
+	Image *string `json:"image,omitempty"`
+
+	// Falcon Container Version. The latest version will be selected when version specifier is missing; ignored when Image is set.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Falcon Container Image Version",order=6
 	Version *string `json:"version,omitempty"`
 }
 
-// Represents the status of Falcon deployment
-type FalconContainerStatusPhase string
+type FalconContainerInjectorSpec struct {
+	// Define annotations that will be passed down to injector service account. This is useful for passing along AWS IAM Role or GCP Workload Identity.
+	// +kubebuilder:default:={name:default}
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Service Account Configuration",order=1
+	ServiceAccount FalconContainerServiceAccount `json:"serviceAccount,omitempty"`
+
+	// +kubebuilder:default:=4433
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:validation:Minimum:=0
+	// +kubebuilder:validation:Maximum:=65535
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Falcon Container Injector Listen Port",order=2
+	ListenPort *int32 `json:"listenPort,omitempty"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Falcon Container Injector TLS Configuration",order=3
+	TLS FalconContainerInjectorTLS `json:"tls,omitempty"`
```

Comment:
This needs to handle custom certs as well as [cert](https://cert-manager.io/) or it will break customer envs